transport/logger.go

```diff
@@ -0,0 +1,69 @@
+package transport
```

Comment:
how does logger relates to transport? It feels like it belongs to root package

Reply:
Almost all logging takes place in the transport package, and putting the entire logger inside of scylla package creates an import cycle, transport package should not have dependencies in scylla package.

I suppose Logger interface could be declared identically in both transport and scylla packages instead of aliasing it in scylla package so it shows up as the whole declaration instead of an alias in docs.

It also could have its own package but that seems excessive to me.